src/charm.py

```diff
@@ -143,6 +159,8 @@ def _on_start(self, event: StartEvent) -> None:
             self.unit.status = BlockedStatus("Failed to initialize juju units operations table")
             return
 
+        self._peers.data[self.app]["units-added-to-cluster"] = "1"
+
```

Comment:
I wonder if this is informative. 
Without understanding too much context, would it be useful to have a dict of unit-names within the application data, added when a unit starts, and then whenever we needed this key, a unit could just count keys in the app data?

Reply:
My thinking being that that way, we could add unit-specific data easily, without adding non-user-helpful information in the relation data.

Reply:
So the reason we are setting data in the peer relation databag is to trigger `relation_changed` events on peers. This is where peers (non-leader) move from `WaitingStatus` (waiting to join cluster) to `ActiveStatus`. I deliberately used a counter instead of a list of unit names to avoid implementing a teardown hook just to remove data from the peer relation databag